flips/20201125-bip-44-multi-account.md

```diff
@@ -5,192 +5,282 @@
+   public key registry. If an address is found, query the Flow network to fetch
+   the account's details. If an account is found, remember the relationship between the
+   path used to generate this key, and the account's details.
+2. Derive a key pair (starting with account index = 0 and key index =
+   1) using the path specified in this FLIP.
+3. Scan for the key usage by checking the public key registry.
+   If the gap limit has been reached for both key index and account index, stop
+   discovery.
+    3.1. If no address is found in the registry 
+        3.1.1 If the key index gap limit has been reached without finding any 
+        addressed in the registry, then go to step 2, incrementing the account index
+        by one and starting with key index = 0 again. 
+        3.1.2 If the key index gap limit has not been reached, go to step 2 and
+        increment the key index by one. 
+    3.2. If an address is found, query the Flow network to fetch the account's details. 
+        3.2.1. If no account is found<sup>1</sup>, go to step 2, incrementing 
+        the account index by one.
+        3.2.2. If an account is found, remember the relationship between the
+        path used to generate this key and the account's details. Then go to step 2, 
+        incrementing the key index by one.
+
+<sup>1</sup>Flow supports account deletion, meaning that an address found in the
+registry may refer to a nonexistent account. In this case the account should be
+skipped but discovery should continue.
+
+#### Account Discovery Conflict Resolution
+
+It is possible for a wallet to incorrectly apply the specifications in this FLIP
+to generate keys.
+
+Should this scenario occur, compliant wallets may be required to work around any
+such previous incorrect usage.
+
+##### Conflict Resolution: Incorrect Use of Account Index
+
+If a wallet has used the account index field incorrectly to generate keys for a
+user, conflict resolution logic may be required when attempting to generate new
+keys.
+
+When generating additional keys, the correct account index to use in the path to
+generate the additional key should be:
+
+- If adding a key to a new account:
+  - The smallest account index which has not yet been used to generate keys
+    set on any account.
+- If adding a key to an existing account:
+  - The largest account index that account discovery determines has been
+    previously used to generate existing keys set on the account, which has also
+    not been used incorrectly to generate keys set on another account.
+  - Otherwise, the smallest account index which has not yet been used to
+    generate keys set on any account.
+
+##### Conflict Resolution: Incorrect Use of Key Index
+
+When generating additional keys for an account, the correct key index to
+use in the path to generate the new key should be the smallest unused key index
+available which has not yet been used to generate a key on the account.
```

Comment:
This is great, and should be enough. 
To avoid confusion we could also add that a path used with at least one curve is considered "used".  This would prevent wallets saying "the path has been used with curve_1, but it hasn't been used with key_2 yet, so I'm gonna use it again".

Reply:
Added a new conflict resolution section for this purpose 👍

Reply:
The new section is great 👌🏼